src/object.rs

```diff
@@ -526,8 +526,16 @@ macro_rules! glib_object_wrapper {
             #[cfg_attr(feature = "cargo-clippy", allow(cast_ptr_alignment))]
             unsafe fn from_glib_borrow(ptr: *mut $ffi_name) -> Self {
                 debug_assert!($crate::types::instance_of::<Self>(ptr as *const _));
-                $name($crate::translate::from_glib_borrow(ptr as *mut _),
-                      ::std::marker::PhantomData)
+                $crate::translate::from_glib_borrow(ptr as *mut $ffi_name)
```

Comment:
The stack overflow comes from here. This impl should've stayed as it was. I was only talking about the new impls in my review comments.

What now happens is that this impl here is calling itself again.